dep prune: panics when Gopkg.lock is deleted

I just deleted and executed `dep prune` and it panicked:

```
$ dep prune
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x13b3aa0]

goroutine 1 [running]:
main.(*pruneCommand).Run(0x16c96a8, 0xc4201317c0, 0xc42000c2e0, 0x0, 0x0, 0x0, 0x0)
	/Users/ahmetb/gotools/src/github.com/golang/dep/cmd/dep/prune.go:71 +0x3a0
main.(*Config).Run(0xc420019020, 0xc420019020)
	/Users/ahmetb/gotools/src/github.com/golang/dep/cmd/dep/main.go:159 +0x8b3
main.main()
	/Users/ahmetb/gotools/src/github.com/golang/dep/cmd/dep/main.go:44 +0x253
```

--- a/cmd/dep/prune.go
+++ b/cmd/dep/prune.go
@@ -7,7 +7,6 @@ package main
 import (
 	"bytes"
 	"flag"
-	"fmt"
 
 	"github.com/golang/dep"
 	"github.com/golang/dep/internal/gps"
@@ -68,8 +67,12 @@ func (cmd *pruneCommand) Run(ctx *dep.Ctx, args []string) error {
 		return errors.Wrap(err, "could not set up solver for input hashing")
 	}
 
+	if p.Lock == nil {
+		return errors.Errorf("Gopkg.lock must exist for prune to know what files are safe to remove.")
+	}
+
 	if !bytes.Equal(s.HashInputs(), p.Lock.Memo) {
-		return fmt.Errorf("lock hash doesn't match")
+		return errors.Errorf("lock hash doesn't match")
 	}
 
 	return dep.PruneProject(p, sm)

--- a/cmd/dep/testdata/harness_tests/prune/without_lock/final/Gopkg.toml
+++ b/cmd/dep/testdata/harness_tests/prune/without_lock/final/Gopkg.toml
@@ -1,0 +1,3 @@
+[[dependencies]]
+  name = "github.com/sdboyer/deptest"
+  version = "^0.8.0"

--- a/cmd/dep/testdata/harness_tests/prune/without_lock/initial/Gopkg.toml
+++ b/cmd/dep/testdata/harness_tests/prune/without_lock/initial/Gopkg.toml
@@ -1,0 +1,3 @@
+[[dependencies]]
+  name = "github.com/sdboyer/deptest"
+  version = "^0.8.0"

--- a/cmd/dep/testdata/harness_tests/prune/without_lock/testcase.json
+++ b/cmd/dep/testdata/harness_tests/prune/without_lock/testcase.json
@@ -1,0 +1,6 @@
+{
+  "commands": [
+    ["prune"]
+  ],
+  "error-expected": "Gopkg.lock must exist"
+}

